src/telliot_feeds/queries/abi_query.py

```diff
@@ -36,11 +37,27 @@ def query_data(self) -> bytes:
 
         This method uses ABI encoding to encode the query's parameter values.
         """
-        param_values = [getattr(self, p["name"]) for p in self.abi]
-        param_types = [p["type"] for p in self.abi]
-        encoded_params = encode_abi(param_types, param_values)
+        # If the query has parameters
+        if self.abi:
+            param_values = [getattr(self, p["name"]) for p in self.abi]
+            param_types = [p["type"] for p in self.abi]
+            encoded_params = encode_abi(param_types, param_values)
 
-        return encode_abi(["string", "bytes"], [type(self).__name__, encoded_params])
+            return encode_abi(["string", "bytes"], [type(self).__name__, encoded_params])
+        # If the query has no real parameters, and only the default "phantom" parameter
+        else:
+            # By default, the queries with no real parameters have a phantom parameter with
+            # a consistent value of empty bytes. The encoding of these empty bytese in
+            # Python does not match the encoding in Solidity, so we need this hacky solution
+            # where the empty bytes are encoded, the almost-accurate query data is generated,
+            # converted to a string, a single character is replaced, then the hex string is
+            # converted back to bytes.
+            encoded_params = encode_single("bytes", b"")
+            q_data_hex_str = encode_abi(["string", "bytes"], [type(self).__name__, encoded_params]).hex()
+            q_data_hex_lis = list(q_data_hex_str)
+            q_data_hex_lis[-66] = "2"
+            q_data_hex_str = "".join(q_data_hex_lis)
+            return bytes.fromhex(q_data_hex_str)
```

Comment:
this would give you the same results as the spec.  Also, I don't know but I prefer if you just added an attribute to the subclass like ```empty_bytes: bytes = b"".rjust(32, b"\0"``` instead of editing this file.

Reply:
I used this one you suggested:
```python
if not self.abi: 
   left_side = b'\0 '.rjust(32, b'\0')
   right_side = b'\0'.rjust(32, b'\0')
   encoded_params = left_side + right_side
   
return encode_abi(["string", "bytes"], [type(self).__name__, encoded_params])
```
I agree it would be better to add an attribute to the subclass instead of editing `AbiQuery`, but nothing I've tried has worked, and also none of those other suggestions ^ worked for me.